File: 02._Data_Files/reader_parser.py
import json
import csv
import xml.etree.ElementTree as ET
import yaml
import ruamel.yaml
import sys

def print_separator():
    print('-' * 50)

# Read and parse text file
with open('./me.txt','r',  encoding='utf-8') as file:
    txtContent = file.read()
    print("TXT:")
    print(txtContent)
    print_separator()

# Read and parse XML file
tree = ET.parse('./me.xml')
root = tree.getroot()
xml_data = {}
for child in root:
    if child.tag == 'hobbies':
        xml_data[child.tag] = ', '.join(hobby.text for hobby in child.findall('hobby'))
    else:
        xml_data[child.tag] = child.text
print("XML:")
for key, value in xml_data.items():
    print(f"{key}: {value}")
print_separator()

# Read and parse YAML file with ordered dictionaries
with open('./me.yaml', 'r', encoding='utf-8') as file:
    yamlContent = ruamel.yaml.YAML(typ='safe', pure=True).load(file)

print("YAML:")
for key, value in yamlContent.items():
    if isinstance(value, list):
        value = ', '.join(value)
    print(f"{key}: {value}")
print_separator()


# Read and parse JSON file
with open('./me.json', 'r', encoding='utf-8') as file:
    jsonContent = json.load(file)
    print("JSON:")
    for key, value in jsonContent.items():
        if isinstance(value, list):
            value = ', '.join(value)
        print(f"{key}: {value}")
print_separator()

# Read and parse CSV file
with open('./me.csv', 'r', encoding='utf-8') as file:
    csvReader = csv.reader(file)
    header = next(csvReader)
    print("CSV:")
    for row in csvReader:
        for i, value in enumerate(row):
            print(f"{header[i]}: {value}")
        print()
    print_separator()

File: 02._Data_Files/reader_parser.js
import { readFileSync } from "fs";
import { parseString } from "xml2js";
import yaml from "js-yaml";

function printSeparator() {
    console.log("-".repeat(50));
}

// Read and parse text file
const txtContent = readFileSync("./me.txt", "utf-8");
console.log("\nTXT:");
console.log(txtContent);
printSeparator();

// Read and parse XML file
const xmlContent = readFileSync("./me.xml", "utf-8");
parseString(xmlContent, (err, result) => {
    if (err) {
        console.error("Error parsing XML:", err);
        return;
    }
    const { name, age, hobbies } = result.me;
    console.log("XML:");
    console.log(`name: ${name[0]}`);
    console.log(`age: ${age[0]}`);
    console.log(`hobbies: ${hobbies[0].hobby.join(", ")}`);
});
printSeparator();

// Read and parse YAML
const yamlContent = readFileSync("./me.yaml", "utf-8");
const yamlParsed = yaml.load(yamlContent);
console.log("\nYAML:");
for (const key in yamlParsed) {
    if (Array.isArray(yamlParsed[key])) {
        console.log(`${key}: ${yamlParsed[key].join(", ")}`);
    } else {
        console.log(`${key}: ${yamlParsed[key]}`);
    }
}
printSeparator();

// Read and parse JSON file
const jsonContent = JSON.parse(readFileSync("./me.json", "utf-8"));
console.log("\nJSON:");
for (const key in jsonContent) {
    if (Array.isArray(jsonContent[key])) {
        console.log(`${key}: ${jsonContent[key].join(", ")}`);
    } else {
        console.log(`${key}: ${jsonContent[key]}`);
    }
}
printSeparator();

// Read and parse CSV file
const csvData = readFileSync("./me.csv", "utf-8");
const csvContent = csvData.split(/\r?\n/).map(row => row.split(","));
console.log("\nCSV:");
const headers = csvContent[0];
const data = csvContent[1];
const result = {};
headers.forEach((header, index) => {
    result[header.trim()] = data[index].trim();
});
const hobbiesIndex = headers.indexOf("hobbies");
if (hobbiesIndex !== -1) {
    const hobbies = [];
    for (let i = hobbiesIndex; i < data.length; i++) {
        hobbies.push(data[i].trim());
    }
    result["hobbies"] = `"${hobbies.join(", ").replace(/^"|"$/g, '')}"`;
}
for (const key in result) {
    console.log(`${key}: ${result[key]}`);
}
printSeparator();